docs/releases/upgrading/upgrade-v10.0.md

```diff
@@ -0,0 +1,270 @@
+
+    Always backup your database and files before you try to perform an upgrade.
+
+!!! info "Older versions"
+
+    In case you have an InvenioRDM installation older than v9, you can gradually upgrade using the existing instructions to v9 and afterwards continue from here.
+
+## Upgrade Steps
+
+!!! warning "Upgrade your invenio-cli"
+
+    Make sure you have the latest `invenio-cli`, for InvenioRDM v10 the release is v1.0.6.
+
+### Elasticsearch check
+
+InvenioRDM v10.0 removes support for Elasticsearch v6. If you are using it, you will need **first** to migrate to
+Elasticsearch v7 or directly to OpenSearch, before upgrading
+
+!!! warning "Elasticsearch v7 change of license"
+
+    Do not upgrade to Elasticsearch v7.11, due to the [change of license](https://www.elastic.co/pricing/faq/licensing).
+    The last version is 7.10.2.
+
+#### Elasticsearch v6 to v7
+
+In your InvenioRDM v9, upgrade from Elasticsearch v6 to v7. Upgrade checklist:
+1. Prepare your new Elasticsearch 7 cluster. For local development, make sure that you have also upgraded Elasticsearch in your `docker-services.yml`.
+2. In your instance, edit the `Pipfile` and replace `invenio-app-rdm = {extras = [...  "elasticsearch6"]...` with `elasticsearch7`. For example, in the `Pipfile`:
+    ```bash
+    [packages]
+    -invenio-app-rdm = {extras = ["postgresql", "elasticsearch6"], version = "~=9.1.0"}
+    +invenio-app-rdm = {extras = ["postgresql", "elasticsearch7"], version = "~=9.1.0"}
+    ```
+3. In your instance, edit the `.invenio` file:
+    ```bash
+    -search = elasticsearch6
+    +search = elasticsearch7
+    ```
+4. Do you have any customisation or custom module that contains Elasticsearch v6 mappings?
+   If not, go to the next step. If yes, [review the breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/breaking-changes-7.0.html#breaking_70_indices_changes)
+   and make sure that you copy the `v6` folder and paste it in the same location, with the new name `v7`.
+   The final folder structure should look like this (do **not** forget the `__init__.py` files, they are required!):
+
+    ```bash
+    my-custom-module
+    |-- mappings
+        |-- __init__.py
+        |-- v6
+            |-- __init__.py
+            |-- <record>
+                |-- <record>-v1.0.0.json
+        |-- v7
+            |-- __init__.py
+            |-- <record>
+                |-- <record>-v1.0.0.json
+    ```
+    You can optionally delete the `v6` folder.
+5. Upgrade the Python dependencies:
+    ```bash
+    cd <my-site>
+    pipenv run pip uninstall -y elasticsearch elasticsearch-dsl
+    pipenv run pip install invenio-search[elasticsearch7]
+    ```
+6. Change the `INVENIO_SEARCH_ELASTIC_HOSTS` configuration variable to point to the new cluster. Restart all servers and Celery workers.
+7. Create the indices and re-index all data:
+    ```bash
+    cd <my-site>
+    # create indices
+    pipenv run invenio index init
+
+    # reindex records
```

Comment:
Unfortunately, here we would also need to run the code snippet from line 194. Otherwise some record types won't be re-indexed (i.e. requests, users, etc.). To avoid repetition we can put the snippet at the bottom as an "annex" and then reference it?

Reply:
I removed this section. I realized that there is no way to have `ES 6` on [RDM v9](https://github.com/inveniosoftware/invenio-app-rdm/blob/8bb35ac2f2f8a0b8c6429a425d6b48e1454d0a59/setup.cfg#L76).
We will help in the chat if needed...